src/cast_core/src/QuiesceActor.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright 2021-present MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cast_core/actors/QuiesceActor.hpp>
+
+#include <memory>
+
+#include <yaml-cpp/yaml.h>
+
+#include <bsoncxx/json.hpp>
+
+#include <mongocxx/client.hpp>
+#include <mongocxx/collection.hpp>
+#include <mongocxx/database.hpp>
+
+#include <boost/log/trivial.hpp>
+
+#include <gennylib/Cast.hpp>
+#include <gennylib/context.hpp>
+#include <gennylib/quiesce.hpp>
+
+namespace genny::actor {
+
+struct QuiesceActor::PhaseConfig {
+    PhaseConfig(PhaseContext& context) {
+        auto actorDb = context.actor()["Database"].maybe<std::string>();
+        if (!actorDb) {
+            BOOST_THROW_EXCEPTION(
+                InvalidConfigurationException("Must give Database in Actor block."));
+        }
+        dbName = *actorDb;
```

Comment:
nit but you don't need this error-checking - `Node` does it for you when you call `.to<>()`:

```
PhaseConfig(ctx)
: dbName{context.actor()["Database"].to<string>()} {}
```

will barf with a helpful error message if not present.